src/openrct2/ReplayManager.cpp

```diff
@@ -0,0 +1,796 @@
+            serialiser << gCheatsEnableChainLiftOnAllTrack;
+            serialiser << gCheatsShowAllOperatingModes;
+            serialiser << gCheatsShowVehiclesFromOtherTrackTypes;
+            serialiser << gCheatsFastLiftHill;
+            serialiser << gCheatsDisableBrakesFailure;
+            serialiser << gCheatsDisableAllBreakdowns;
+            serialiser << gCheatsBuildInPauseMode;
+            serialiser << gCheatsIgnoreRideIntensity;
+            serialiser << gCheatsDisableVandalism;
+            serialiser << gCheatsDisableLittering;
+            serialiser << gCheatsNeverendingMarketing;
+            serialiser << gCheatsFreezeWeather;
+            serialiser << gCheatsDisablePlantAging;
+            serialiser << gCheatsAllowArbitraryRideTypeChanges;
+            serialiser << gCheatsDisableRideValueAging;
+            serialiser << gConfigGeneral.show_real_names_of_guests;
+            serialiser << gCheatsIgnoreResearchStatus;
+
+            return true;
+        }
+
+        bool SerialiseCommand(DataSerialiser& serialiser, ReplayCommand& command)
+        {
+            serialiser << command.tick;
+            serialiser << command.commandIndex;
+
+            bool isGameAction = false;
+            if (serialiser.IsSaving())
+            {
+                isGameAction = command.action != nullptr;
+            }
+            serialiser << isGameAction;
+
+            if (isGameAction)
+            {
+                uint32_t actionType = 0;
+                if (serialiser.IsSaving())
+                {
+                    actionType = command.action->GetType();
+                }
+                serialiser << actionType;
+
+                if (serialiser.IsLoading())
+                {
+                    command.action = GameActions::Create(actionType);
+                    Guard::Assert(command.action != nullptr);
+                }
+
+                command.action->Serialise(serialiser);
+            }
+            else
+            {
+                serialiser << command.eax;
+                serialiser << command.ebx;
+                serialiser << command.ecx;
+                serialiser << command.edx;
+                serialiser << command.esi;
+                serialiser << command.edi;
+                serialiser << command.ebp;
+                serialiser << command.callback;
+            }
+            return true;
+        }
+
+        bool Serialise(DataSerialiser& serialiser, ReplayRecordData& data)
+        {
+            serialiser << data.magic;
+            if (data.magic != ReplayMagic)
+            {
+                log_error("Magic does not match %08X", data.magic);
+                return false;
+            }
+            serialiser << data.version;
+            if (data.version != ReplayVersion)
+            {
+                log_error("Invalid version detected %04X", data.version);
+                return false;
+            }
+            serialiser << data.networkId;
```

Comment:
add `log_warning` here which prints actual and expected version if mismatched and continue execution

Reply:
Just so that I understand this correctly, you wan't me to remove the return and instead put a warning there? I don't think we should attempt parse an unknown format, if we ever change the version number we should branch the path here.

Reply:
No, leave the error and return for recording format version, but add a warning for mismatched network version

Reply:
Gotcha